python/src/aac/context/definition_parser.py

```diff
@@ -9,735 +9,841 @@
                 raise LanguageError(
-                    f"Failed to create Enum instance_class for {enum_definition.name}: {e.message}",
-                    get_location_str(enum_definition.name, enum_definition.lexemes),
+                    f"Failed to create instance_class for {schema_definition.name}: {e.message}",
+                    self.get_location_str(
+                        schema_definition.name, schema_definition.lexemes
+                    ),
                 )
-            context.context_instance.fully_qualified_name_to_class[
-                fully_qualified_name
-            ] = instance_class
-            return instance_class
-
-        def create_schema_class(schema_definition: Definition) -> Type:
-            """
-            Creates a schema class from a given schema definition.
-
-            Args:
-                schema_definition (Definition): A schema definition to convert to a class.
-
-            Returns:
-                The created class.
-            """
-            instance_class = None
-
-            fully_qualified_name = schema_definition.get_fully_qualified_name()
-            if schema_definition.get_root_key() == "primitive":
-                # this is a primitive, so there's no structure to create...just return the python type
-                return eval(primitive_name_to_py_type[schema_definition.name])
-            elif schema_definition.get_root_key() == "enum":
-                # this is an enum, so create the enum class
-                return create_enum_class(schema_definition)
-            elif schema_definition.get_root_key() != "schema":
+        else:
+            parent_classes = inheritance_parents  # the following causes a method resolution order (MRO) error when creating the type: [object] + inheritance_parents
+            try:
+                instance_class = type(
+                    schema_definition.get_python_class_name(),
+                    tuple(parent_classes),
+                    {"__module__": schema_definition.get_python_module_name()},
+                )
+            except LanguageError as e:
                 raise LanguageError(
-                    f"Definition {schema_definition.name} is not a schema",
-                    get_location_str(
-                        schema_definition.get_root_key(), schema_definition.lexemes
+                    f"Failed to create instance_class for {schema_definition.name}: {e.message}",
+                    self.get_location_str(
+                        schema_definition.name, schema_definition.lexemes
                     ),
                 )
+        return instance_class
+
+    def create_schema_class(self, schema_definition: Definition) -> Type:
+        """
+        Creates a schema class from a given schema definition.
+
+        Args:
+            schema_definition (Definition): A schema definition to convert to a class.
+
+        Returns:
+            The created class.
+        """
+        instance_class = None
```

Comment:
```suggestion
```
`instance_class` is declared to `None` again at line 319, since it is not used until after 319, shouldn't need to have it declared at the start of the method and can keep it grouped with where it is utilized.